src/routes/settings/ManageFederations.tsx

```diff
@@ -107,50 +143,229 @@ function AddFederationForm(props: { refetch?: RefetchType }) {
+                                                                label={false}
+                                                            />
+                                                        )}
+                                                    </For>
+                                                </div>
+                                            </KeyValue>
+                                        </Show>
+                                        {/* FIXME: do something smarter than just taking first code */}
+                                        <Button
+                                            intent="blue"
+                                            onClick={() =>
+                                                onSelect(fed.invite_codes[0])
+                                            }
+                                            loading={fed.invite_codes.includes(
+                                                loadingFederation()
+                                            )}
+                                        >
+                                            {i18n.t(
+                                                "settings.manage_federations.add"
+                                            )}
+                                        </Button>
+                                    </VStack>
+                                </FancyCard>
                             )}
-                            placeholder="fed11..."
-                            required
-                        />
-                    )}
-                </Field>
+                        </For>
+                    </Match>
+                </Switch>
+            </Suspense>
+        </>
+    );
+}
+
+function RecommendButton(props: { fed: MutinyFederationIdentity }) {
+    const [state] = useMegaStore();
+    const [recommendLoading, setRecommendLoading] = createSignal(false);
+    // This is just some local state that makes it feel like they've recommended it
+    // even if they aren't a "real person"
+    const [recommended, setRecommended] = createSignal(false);
+
+    const [recommendedByMe, { refetch }] = createResource(async () => {
+        try {
+            const federations: DiscoveredFederation[] =
+                await state.mutiny_wallet?.discover_federations();
+            const thisFederation = federations.find(
+                (f) => f.id === props.fed.federation_id
+            );
+            if (!thisFederation) {
+                return undefined;
+            } else {
+                const myNpub = state.mutiny_wallet?.get_npub();
+                const myHexpub = await hexpubFromNpub(myNpub);
+                const myRecommend = thisFederation.recommendations.find(
+                    (r) => r.npub === myHexpub
+                );
+                return myRecommend;
+            }
+        } catch (e) {
+            console.error(e);
+            return undefined;
+        }
+    });
```

Comment:
I'll add a function to mutiny-node that does this more efficiently, but shouldn't be blocking